fedimint-core/src/module/audit.rs

```diff
@@ -52,18 +56,133 @@ impl Display for Audit {
         for item in &self.items {
             formatter.write_fmt(format_args!("\n{item}"))?;
         }
-        formatter.write_fmt(format_args!("\n{}", self.sum()))
+        formatter.write_fmt(format_args!("\n{}", self.net_assets()))
     }
 }
 
 pub struct AuditItem {
     pub name: String,
     pub milli_sat: i64,
+    pub module_name: String,
 }
 
 impl Display for AuditItem {
     fn fmt(&self, formatter: &mut Formatter) -> std::fmt::Result {
         let sats = (self.milli_sat as f64) / 1000.0;
-        formatter.write_fmt(format_args!("{:>+15.3}|{}", sats, self.name))
+        formatter.write_fmt(format_args!(
+            "{:>+15.3}|{:>10}|{}",
+            sats, self.module_name, self.name
+        ))
     }
 }
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq)]
+pub struct AuditSummary {
+    pub net_assets: i64,
+    pub module_summaries: HashMap<String, ModuleSummary>,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq)]
+pub struct ModuleSummary {
+    pub net_assets: i64,
+}
+
+impl AuditSummary {
+    pub fn from_audit(audit: &Audit) -> Self {
+        AuditSummary {
+            net_assets: calculate_net_assets(audit.items.iter().collect()),
+            module_summaries: generate_module_summaries(&audit.items),
+        }
+    }
+}
+
+fn generate_module_summaries(audit_items: &[AuditItem]) -> HashMap<String, ModuleSummary> {
+    audit_items
+        .iter()
+        .map(|item| (item.module_name.clone(), item))
+        .into_group_map()
+        .into_iter()
+        .map(|(module_name, module_audit_items)| {
+            (
+                module_name,
+                ModuleSummary {
+                    net_assets: calculate_net_assets(module_audit_items),
+                },
+            )
+        })
+        .collect()
+}
+
+fn calculate_net_assets(items: Vec<&AuditItem>) -> i64 {
```

Comment:
nit: You don't need a `Vec` (owned data structure) here, especially if it contains references. I'd go with a slice `&[AuditItem]` or an iterator `impl Iterator<Item=&AuditItem>`.